src/utilities/ClearCaches.php

```diff
@@ -184,7 +184,7 @@ public static function cacheOptions(): array
                     }
 
                     $basePath = Craft::getAlias($basePath);
-                    if ($basePath !== false) {
+                    if ($basePath !== false && file_exists($basePath)) {
```

Comment:
Checking `file_exists` and not `is_dir`, so that you _will_ get an error if `\craft\config\GeneralConfig::$resourceBasePath` is pointing to a non-dir file.